climada/entity/exposures/litpop/litpop.py

```diff
@@ -69,16 +68,21 @@ class LitPop(Exposures):
     """
     _metadata = Exposures._metadata + ['exponents', 'fin_mode', 'gpw_version']
 
-    def set_countries(self, countries, res_arcsec=30, exponents=(1,1),
-                      fin_mode='pc', total_values=None, admin1_calc=False,
-                      reference_year=DEF_REF_YEAR, gpw_version=GPW_VERSION,
-                      data_dir=SYSTEM_DIR):
+    def set_countries(self, *args, **kwargs):
+        """This function is deprecated, use LitPop.from_countries instead."""
+        LOGGER.warning("The use of LitPop.set_countries is deprecated."
+                       "Use LitPop.from_countries instead.")
+        self.__dict__ = LitPop.from_countries(*args, **kwargs).__dict__
+
+    @classmethod
```

Comment:
In this case, I decided to use a `@classmethod` so that we have a reference `cls` to the class that the developer is using to call this constructor. In the future, if somebody implements a derived class of `LitPop` (say `AdvancedLitPop`), this method can still be used as `AdvancedLitPop.from_countries(...)` and will return an instance of `AdvancedLitPop` instead of `LitPop`.

In some cases, the usage of `@staticmethod` might make sense, but in general, a `@classmethod` is safer.

Reply:
What would be reasons / cases to use `@staticmethod`?

Reply:
That would be a hypothetical case, where the `set_`-method is not really a pure constructor intrinsically, but does something more involved. I haven't looked at all the `set_` methods in detail, but some of them might be different from this example. They might create a new object from a given object or from a list of objects, like `Hazard.concat`. In that case the type of the returned object is determined using input parameters instead of the `cls` argument.